assets/src/components/v2/departures/section.ts

```diff
@@ -106,37 +123,88 @@ const trimOneBy = (
 // Destructively shift the last above-the-fold departure time in a section to
 // below-the-fold, if possible. Returns `true` if the section was trimmed.
 const trimSection = (section: FoldedSection): boolean => {
-  if (section.type == "folded_section") {
-    const {
-      rows: { aboveFold, belowFold },
-    } = section;
-
-    for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
-      const row = aboveFold[aboveIndex];
-
-      if (row.type == "departure_row") {
-        const trimmedTimeWithCrowding = _.last(row.times_with_crowding);
-
-        if (row.times_with_crowding.length > 1) {
-          // More than one departure time; trim the last one.
-          row.times_with_crowding.splice(row.times_with_crowding.length - 1);
-        } else {
-          // Only one departure time; trim this row entirely.
-          aboveFold.splice(aboveIndex, 1);
-        }
-
-        if (trimmedTimeWithCrowding != null) {
-          belowFold.unshift({
-            ...row,
-            id: _.uniqueId(`${row.id}_`),
-            times_with_crowding: [trimmedTimeWithCrowding],
-          });
-        }
-
-        return true;
+  const {
+    rows: { aboveFold, belowFold },
+  } = section;
+
+  for (let aboveIndex = aboveFold.length - 1; aboveIndex >= 0; aboveIndex--) {
+    const row = aboveFold[aboveIndex];
+
+    if (isDepartureRow(row)) {
+      const timesWithCrowding = row.times_with_crowding;
+      const trimmedTimeWithCrowding = _.last(timesWithCrowding);
+
+      if (timesWithCrowding.length > 1) {
+        // More than one departure time; trim the last one.
+        timesWithCrowding.splice(timesWithCrowding.length - 1);
+      } else {
+        // Only one departure time; trim this row entirely.
+        aboveFold.splice(aboveIndex, 1);
+      }
+
+      if (trimmedTimeWithCrowding != null) {
+        belowFold.unshift({
+          ...row,
+          id: _.uniqueId(`${row.id}_`),
+          times_with_crowding: [trimmedTimeWithCrowding],
+        });
       }
+
+      return true;
     }
   }
 
   return false;
 };
+
+const trimSectionForDestinationGrouping = (
+  section: FoldedDepartureGroupedSection,
+): boolean => {
+  const {
+    rows: { aboveFold },
+    direction_to_trim,
+  } = section;
+
+  // If there's only one direction, treat it as a timed list and trim the furthest time
```

Comment:
The comment here is now out-of-sync with the code, which no longer has this conditional. I think it can just be removed.